perl_genrule.bzl

```diff
@@ -0,0 +1,128 @@
+    out_files.append(out_file)
+    return commands, src_files, out_files
+
+def generate_commands(binary, assembly_flavor, srcs_to_outs, srcs_to_outs_dupes, ctx):
+    """Generate commands needed to produces outs from sources. 
+
+    Args:
+        binary: The binary to run
+        assembly_flavor: The type of assembly to produce
+        srcs_to_outs: The main sources to outputs dict
+        srcs_to_outs_dupes: The secondary sources to outputs dict
+        ctx: The bazel rule context
+    Returns:
+        The commands joined on comma, the source files and the output files
+    """
+
+    commands = []
+    out_files = []
+    src_files = []
+    for src, out in srcs_to_outs.items():
+        intermediate_commands, intermediate_src_files, intermediate_out_files = generate_single_command(binary, assembly_flavor, src, out, ctx)
+        commands = commands + intermediate_commands
+        out_files = out_files + intermediate_out_files
+        src_files = src_files + intermediate_src_files
+    for src, out in srcs_to_outs_dupes.items():
+        intermediate_commands, intermediate_src_files, intermediate_out_files = generate_single_command(binary, assembly_flavor, src, out, ctx)
+        commands = commands + intermediate_commands
+        out_files = out_files + intermediate_out_files
+        src_files = src_files + intermediate_src_files
+    if ctx.attr.is_unix:
+        return ",".join(commands), src_files, out_files
+    else:
+        return ";".join(commands), src_files, out_files
+
+def _perl_genrule_impl(ctx):
+    # On Unix we want to use rules_perl version
+    binary_invocation = "perl"
+    if ctx.attr.is_unix:
+        binary_invocation = ctx.toolchains["@rules_perl//perl:toolchain_type"].perl_runtime.interpreter.path
+    additional_srcs = combine_list_of_lists([src.files.to_list() for src in ctx.attr.additional_srcs])
+
+    commands_joined, srcs_as_files, outs_as_files = generate_commands(binary_invocation, ctx.attr.assembly_flavor, ctx.attr.srcs_to_outs, ctx.attr.srcs_to_outs_dupes, ctx)
+    outs_as_files_paths = [out.path for out in outs_as_files]
+    srcs_as_files_paths = [src.path for src in srcs_as_files]
+    perl_generate_file = ctx.file._perl_generate_file
+    if ctx.attr.is_unix:
+        ctx.actions.run(
+            inputs = srcs_as_files + additional_srcs + [ctx.toolchains["@rules_perl//perl:toolchain_type"].perl_runtime.interpreter],
+            outputs = outs_as_files,
+            executable = perl_generate_file,
+            arguments = [commands_joined],
+            mnemonic = "GenerateAssemblyFromPerlScripts",
+            progress_message = "Generating files {} from scripts {}".format(outs_as_files_paths, srcs_as_files_paths),
+        )
+    else:
+        ctx.actions.run_shell(
+            inputs = srcs_as_files + additional_srcs,
+            outputs = outs_as_files,
+            command = commands_joined,
+            mnemonic = "GenerateAssemblyFromPerlScriptsOnWindows",
```

Comment:
`rules_perl` should support windows as of `0.3.0`, is this still needed?

Reply:
I believe this can be made to work, but would require either replacing or augmenting `perl_generate_file.sh` with something cross-platform - PRs welcome :)